archive_communication/api-requirements.md

```diff
@@ -0,0 +1,324 @@
+    <td class="tg-lboi">Should have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>Preferred OAuth mechanism would be ORCID.</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">2</td>
+    <td class="tg-lboi">Service account authentication</td>
+    <td class="tg-lboi">A downstream service can authenticate with an API Key to automate interactions with the API.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">3</td>
+    <td class="tg-lboi">Consistent, persistent identifier for objects</td>
+    <td class="tg-lboi">A user or service should be able to access an object using the same identifying URI each time.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">4</td>
+    <td class="tg-lboi">Object schema documentation</td>
+    <td class="tg-lboi">Objects fetched from API contain links to the schema used to generate/validate objects.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>JSONSchema preferred</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">5</td>
+    <td class="tg-lboi">Versioned data schemas</td>
+    <td class="tg-lboi">Schemas should have versions so as the schemas change, we can track which schema was used to generate/validate objects.</td>
+    <td class="tg-lboi">Should have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>JSONSchema preferred</li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">6</td>
+    <td class="tg-lboi">Related objects are easily findable from a given object</td>
+    <td class="tg-lboi">Objects contain resource IDs or resource URIs for related objects.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li> Supports use case <a href="api-use-cases.md"> Data inventory – new project</a>
+            <li><a href="#related-objects">See example</a></li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">7</td>
+    <td class="tg-lboi">Consistent response status codes</td>
+    <td class="tg-lboi">API responses should contain appropriate, standardized status codes.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi">
+        <ul>
+            <li>
+                <a href="https://datatracker.ietf.org/doc/html/rfc2616"> 
+                    Proposed standard: IETF RFC 2616
+                </a>
+            </li>
+        </ul>
+    </td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">8</td>
+    <td class="tg-lboi">API endpoint documentation</td>
+    <td class="tg-lboi">Public endpoints should be documented with expected request params and sample responses.</td>
+    <td class="tg-lboi">Must have</td>
+    <td class="tg-lboi"></td>
+  </tr>
+  <tr>
+    <td class="tg-lboi">9</td>
```

Comment:
@pbishwakarma this will only work for the `search` API that is under current development for the Brain Image Library